cmd/api/src/database/sso_providers.go

```diff
@@ -69,3 +67,40 @@ func (s *BloodhoundDB) GetAllSSOProviders(ctx context.Context, order string, sql
 	result := query.Find(&providers)
 	return providers, CheckError(result)
 }
+
+// CreateSSOProvider creates an entry in the sso_providers table
+// A slug will be created for the SSO Provider using the name argument as a base. The name will be lower cased and all spaces are replaced with `-`
+func (s *BloodhoundDB) CreateSSOProvider(ctx context.Context, name string, authProvider model.SessionAuthProvider) (model.SSOProvider, error) {
+	provider := model.SSOProvider{
+		Name: name,
+		Slug: strings.ToLower(strings.ReplaceAll(name, " ", "-")),
+		Type: authProvider,
+	}
+
+	return provider, CheckError(s.db.WithContext(ctx).Table(ssoProviderTableName).Create(&provider))
+}
+
+// DeleteSSOProvider deletes a sso_provider entry with a matching id
+func (s *BloodhoundDB) DeleteSSOProvider(ctx context.Context, id int) error {
+	var (
+		ssoProvider, err = s.GetSSOProvider(ctx, id)
+		auditEntry       = model.AuditEntry{
+			Action: "DeleteSSOProvider",
```

Comment:
Can we add this string to the constant definition in `bhce/cmd/api/src/model/audit.go` please?